### ISSUE TITLE
SQL error when saving new cart using Statamic Eloquent Driver

### ISSUE THREAD

**Issue by Ririshi** [doublethreedigital/simple-commerce#486]
### Bug Description

I just set up a new project, installing the [Statamic Eloquent Driver](https://github.com/statamic/eloquent-driver) using the incrementing ids option. After that, I installed the Simple Commerce starter kit. After creating a homepage and a cart page (in the Eloquent entry storage), I opened the cart page, which resulted in an SQL error.

The newly created cart (`Order` object) has a UUID generated by `Stache::generateId()`, but the `EntryModel` used by the Eloquent Driver assumes it will be saving an integer id. The MySQL database backing this storage obviously has an incrementing id column, rather than a UUID column, so that goes wrong.

Sadly, I need to get this project up and running asap, so I opted to use the `UuidEntryModel` driver model as a workaround. I did not copy the exact error I was getting while trying to set it up originally, but it boils down to trying to save a string (UUID) in an integer column.

### Steps to reproduce

1. Get a fresh install of statamic/statamic
2. Install Statamic Eloquent Driver using their "fresh install"
3. Install the Simple Commerce starter kit (or simply install SC normally, I suppose)
4. Create a homepage and a cart page (also when using the starter kit, since the kit only creates new file-based entries)
5. Attempt to open the cart page, which will result in a SQL error.

### Environment

```
Statamic 3.2.18 Pro
Laravel 8.68.0
PHP 7.4.3
doublethreedigital/simple-commerce 2.3.49
statamic/eloquent-driver 0.2.0
```

**Comment by duncanmcclean** [comment]
I unfortunately don't have time to look into this tonight. The soonest I'd be able to investigate is tomorrow evening.

Does using the `UuidEntryModel` work for you in the time being or does that also give you errors?

**Comment by duncanmcclean** [comment]
Alright, managed to find some time to identify & fix the issue you pointed out but it looks like there are other bugs when using the Eloquent driver with SC. Things like not being able to view things you've added to the cart (multiple carts seem to be generated). 

I don't have enough time to fix that tonight but I've opened a draft PR which I'll loop back to when I have some time: #487.

In the mean time, I'd recommend using the `UuidEntryModel` (if that works for you) or using flat files for your entries if that's possible? 

At some point in the future, SC will have it's own set of Eloquent models so you can use a database if you need to but I don't have time to implement that at the moment.

**Comment by duncanmcclean** [comment]
Fixed in latest release - [v2.3.50](https://github.com/doublethreedigital/simple-commerce/releases/tag/v2.3.50)
